Java/libraries/recognizers-text-number/src/main/java/com/microsoft/recognizers/text/number/parsers/BaseNumberParser.java

```diff
@@ -419,9 +419,9 @@ private boolean skipNonDecimalSeparator(char ch, int distance, CultureInfo cultu
         // Special cases for multi-language countries where decimal separators can be used interchangeably. Mostly informally.
         // Ex: South Africa, Namibia; Puerto Rico in ES; or in Canada for EN and FR.
         // "me pidio $5.00 prestados" and "me pidio $5,00 prestados" -> currency $5
-        Pattern cultureRegex = Pattern.compile("^(en|es|fr)(-)?\b", Pattern.CASE_INSENSITIVE | Pattern.UNICODE_CHARACTER_CLASS);
+        Pattern cultureRegex = Pattern.compile("^(en|es|fr)(-)?\\b", Pattern.CASE_INSENSITIVE | Pattern.UNICODE_CHARACTER_CLASS);
```

Comment:
There were two issues preventing the appropriate parsing of "3.5ans".
1. The first is that in this regex, "\b" was used, but in Java this is an escape code for a backspace, and does not correspond to a word boundary.
2. The `matches` method was used even though it seems only a substring match was desired. `matches` requires the whole string to match. Changing this to `find` returns true for "fr-fr" and causes the whole return expression to evaluate to false.

Reply:
Thanks for catching this. Can you add a `@TODO` comment here calling attention to refactor it?
This language list should not be hardcoded there. Ideally it should come from a project config-level file so all platforms are always in sync.